src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/SuccessfulRetryTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace ServiceControlCompatibilityTests
+{
+    using Autofac;
+    using NServiceBus;
+    using NUnit.Framework;
+    using System;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class SuccessfulRetryTests : SqlScTest
+    {
+        [TestCaseSource(nameof(AllTransports))]
+        public async Task Can_successfully_retry_a_failed_message(Type transportDetailType)
+        {
+            var transportDetails = StartUp(transportDetailType);
+            await RunTest(transportDetails);
+        }
+
+        async Task RunTest(ITransportDetails transportDetails)
+        {
+            var handlerInstance = new TestMessageHandler();
+            var builder = new ContainerBuilder();
+            builder.RegisterInstance(handlerInstance);
+            var container = builder.Build();
```

Comment:
I have an idea for this but my VS has been taken out of commission by an incredibly slow update today.

Something like this?

```
var endpointDetails = new EndpointDetails(new TestContext { ShouldFail = true });
endpointDetails.Add<TestMessageHandler>();

var endpointA = await CreateSqlEndpoint("EndpointA", transportDetails);
var endpointB = await CreateSqlEndpoint("EndpointB", transportDetails, endpointDetails);

endpointA.Send(endpointB, new TestMessage());

var failedMessage = await ServiceControl.WaitForNewFailingMessages(endpointA);

endpointDetails.Context.ShouldFail = false;

await ServiceControl.RetryMessageId(failedMessage.Id);

var handled = endpointDetails.Context.WaitForDone().Wait(60000);

Assert.IsTrue(handled, "blah blah");



class TestContext
{
  TaskCompletionSource<bool> doneSource = new TaskCompletionSource<bool>();

  public bool ShouldFail { get; set; }
  
  public void MarkSuccess()
  {
    doneSource.SetResult(true);
  }

  public Task WaitForDone()
  {
    return doneSource.Task;
  }
}

class TestMessageHandler : IHandleMessages<TestMessage>
{
  public TestMessageHandler(TestContext testContext)
  {
    this.testContext = testContext;
  }

  public Task Handle(TestMessage message, IMessageHandlerContext context)
  {
    if(testContext.ShouldFail)
    {
      throw new ApplicationException("Failing...");
    }

    testContext.MarkSuccess();
    return Task.FromResult(0);
  }

  private TestContext testContext;
}

class EndpointDetails<TContext>
{
  public EndpointDetails(TContext context)
  {
    Context = context;
  }

  public void Add<T>()
  {
    // capture T in a list somewhere
  }

  public IContainer CreateContainer()
  {
    var builder = new ContainerBuilder();
    builder.RegisterInstance(Context);
    // For each of the items added above by `Add`, add them to the container
    return builder.Build();
  }

  public TContext Context { get; }
  
}
```